python/setup.py

```diff
@@ -111,12 +127,54 @@ def run(self):
         build_py.run(self)
 
 
-class CustomBuildExt(build_ext):
+class BuildGLExt(build_ext):
     def run(self):
+        self.extensions = [ext for ext in self.extensions if ext.name == GL_EXT_NAME]
         self.run_command("build_proto")
         build_ext.run(self)
 
 
+class BuildGLTorchExt(torch.utils.cpp_extension.BuildExtension if torch else build_ext):
+    def run(self):
+        assert (
+            torch
+        ), "Building graphlearn-torch extension requires installing pytorch first. Let WITH_GLTORCH=OFF if you don't need it."
+        self.extensions = [
+            ext
+            for ext in self.extensions
+            if ext.name in [GLTORCH_EXT_NAME, GLTORCH_V6D_EXT_NAME]
+        ]
+        torch.utils.cpp_extension.BuildExtension.run(self)
+
+    def _get_gcc_use_cxx_abi(self):
+        if hasattr(self, "_gcc_use_cxx_abi"):
+            return self._gcc_use_cxx_abi
+        output = subprocess.run(
+            [shutil.which("cmake"), "."],
+            cwd=glt_root_path,
+            capture_output=True,
+            text=True,
+        )
+        import re
```

Comment:
Maybe running cmake in a clean build directory would be better, i.e.,

```
build_dir = os.path.join(glt_root_path, 'cmake-build')
os.makedirs(build_dir, exist_ok=True)
output = subprocess.run([shutil.which('cmake'), '..'], cwd=build_dir, ....)
```

Reply:
Fixed by adding a clean cmake build directory.